### PR TITLE
cmd: Take sensitive defaults from environment variables

### PR DESCRIPTION
**Pull request by jtumber-ibm**
Adds `os.Getenv` option to each sensitive field.

Fixes: #83

Signed-off-by: James Tumber <james.tumber@ibm.com>

### CODE REVIEW

File: cmd/cloud-api-adaptor/main.go [hunk 1]
```diff
@@ -79,8 +79,8 @@ func (cfg *daemonConfig) Setup() (cmd.Starter, error) {
 
 	case "azure":
 		cmd.Parse("azure", os.Args[1:], func(flags *flag.FlagSet) {
-			flags.StringVar(&azurecfg.ClientId, "clientid", "", "Client Id")
-			flags.StringVar(&azurecfg.ClientSecret, "secret", "", "Client Secret")
+			flags.StringVar(&azurecfg.ClientId, "clientid", os.Getenv("AZURE_CLIENT_ID"), "Client Id")
+			flags.StringVar(&azurecfg.ClientSecret, "secret", os.Getenv("AZURE_CLIENT_SECRET"), "Client Secret")
 			flags.StringVar(&azurecfg.TenantId, "tenantid", "", "Tenant Id")
```
Comment (stevenhorsman) [inline cmd/cloud-api-adaptor/main.go:84]: Hey JT - from https://github.com/confidential-containers/cloud-api-adaptor/pull/284/files it looks like `tenantid` might also be a sensitive field for Azure

**Comment by jtumber-ibm** [comment]
Changed the approach because the sensitive values would be leaked in the command help e.g.
```
  -api-key IBMCLOUD_API_KEY
        IBM Cloud API key (default "my-api-key")
```
Unlikely that someone would be able to exploit that, if they already have access then they can just `echo $IBMCLOUD_API_KEY` but it could be accidentally left in a log during troubleshooting.
Moved the assignment out of the parsing to avoid that.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks for the update!

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @jtumber-ibm

**Pull request merged**